lib/resources/postgres.rb

```diff
@@ -71,6 +94,27 @@ def to_s
 
     private
 
+    def conf_dir_from_process
+      data_dir_command = inspec.command("ps aux | grep 'postgres *-D' | awk '{print $NF}'").stdout.strip
+      if data_dir_command.empty?
+        warn "PostgreSQL process not found - using filesystem to try and determine the pg 'data_dir'"
+        return nil
+      else
+        return data_dir_command
+      end
+    end
+
+    def version_from_process(pg_data_dir)
+      if !pg_data_dir.empty?
+        version_command = inspec.command("cat `find #{pg_data_dir} \\\( ! -path #{pg_data_dir} -prune \\\) -type f -name \"PG_VERSION\"\`").stdout.strip
```

Comment:
I'd rather see us have one `command` resource return the full path to the PG_VERSION file that can then be used in a `file` resource so that we can properly catch exceptions, etc. Using `cat` obfuscates that ability.